src/staking-router/staking-router.service.ts

```diff
@@ -13,6 +16,141 @@ export class StakingRouterService {
     protected readonly keysApiService: KeysApiService,
   ) {}
 
+  /**
+   * Return staking module data and block information
+   */
+  public async getVettedAndUnusedKeys() {
+    // TODO: add cache by modules nonce
+    const { operatorsByModules, unusedKeys, blockHash, blockNumber } =
+      await this.getOperatorsAndKeysFromKAPI();
+    // all staking modules list
+    const stakingModulesData: StakingModuleData[] = operatorsByModules.data.map(
+      ({ operators, module: stakingModule }) => {
+        const { moduleUnusedKeys, moduleVettedKeys } =
+          this.getModuleOperatorsVettedKeys(operators, unusedKeys.data);
+
+        return {
+          unusedKeys: moduleUnusedKeys.map((srKey) => srKey.key),
+          nonce: stakingModule.nonce,
+          stakingModuleId: stakingModule.id,
+          blockHash,
+          vettedKeys: moduleVettedKeys,
+        };
+      },
+    );
+
+    return {
+      stakingModulesData,
+      blockHash,
+      blockNumber,
+    };
+  }
+
+  /**
+   * Request grouped by modules operators and all staking modules keys with meta from KAPI
+   */
+  private async getOperatorsAndKeysFromKAPI() {
+    const operatorsByModules =
+      await this.keysApiService.getOperatorListWithModule();
+    const { blockHash: operatorsBlockHash, blockNumber: operatorsBlockNumber } =
+      operatorsByModules.meta.elBlockSnapshot;
+
+    const unusedKeys = await this.keysApiService.getUnusedKeys();
+    const { blockHash: keysBlockHash } = unusedKeys.meta.elBlockSnapshot;
+
+    this.validateBlockHashMatch(keysBlockHash, operatorsBlockHash);
+
+    return {
+      operatorsByModules,
+      unusedKeys,
+      blockHash: operatorsBlockHash,
+      blockNumber: operatorsBlockNumber,
+    };
+  }
+
+  private validateBlockHashMatch(
+    keysBlockHash: string,
+    operatorsBlockHash: string,
+  ) {
+    if (keysBlockHash !== operatorsBlockHash) {
+      this.logger.error(
+        'Blockhash of the received keys and operators dont match',
+        {
+          keysBlockHash,
+          operatorsBlockHash,
+        },
+      );
+
+      throw new Error(
```

Comment:
Specify Error.
smth like `throw new InconsistentDataError(`
And in `getOperatorsAndKeysFromKAPI` method if you catch this error try to refetch first request

Reply:
You can do it few times until get Consistent data or maxRetryExceed

Reply:
add class to error i can 
retry i will try to consider, thank you